Use arrow functions when possible

teoli2003 opened the pull request:
When possible, we want to use arrow functions for anonymous functions

teoli2003 committed:
Use arrow functions

teoli2003 committed:
Remove polyfill

teoli2003 committed:
Remove polyfill

teoli2003 committed:
Remove polyfill

teoli2003 committed:
Keep function (IE-only feature)

teoli2003 committed:
Fix indentation

teoli2003 committed:
Fix indentation

teoli2003 committed:
Fix over-correction

teoli2003 committed:
Remove over-correction

teoli2003 committed:
Fix syntax

teoli2003 committed:
Remove function

teoli2003 committed:
Fix syntax

teoli2003 committed:
Fix over-correction

teoli2003 committed:
Update files/en-us/web/api/eventtarget/addeventlistener/index.md

Co-authored-by: rubiesonthesky <2591240+rubiesonthesky@users.noreply.github.com>

teoli2003 committed:
Change for arrow function

teoli2003 committed:
Update files/en-us/web/api/htmlformelement/index.md

Co-authored-by: rubiesonthesky <2591240+rubiesonthesky@users.noreply.github.com>

Josh-Cena committed:
Update index.md

Josh-Cena commented on files/en-us/web/api/domtokenlist/foreach/index.md:
```diff
@@ -62,7 +62,7 @@ const classes = pre.classList;
 
 classes.forEach(
-  function(value, key, listObj) {
+  (value, key, listObj) => {
```
This would no longer create the `this` binding, which is passed below.

teoli2003 replied:
Fixed below (please double check)

Josh-Cena replied:
@teoli2003 I don't think that fixes it—we have explicitly passed `"arg"` as the `thisArg`, so in the original example, it would log `arg` as the value of `this` inside the function. Can't say how useful that is in practice, but that's the point of this example.

teoli2003 replied:
Let's revert to a function then?

Josh-Cena replied:
Yes, let's revert it.

teoli2003 replied:
Done.

Josh-Cena committed:
Update index.md

Josh-Cena commented on files/en-us/web/api/event/preventdefault/index.md:
```diff
@@ -143,10 +143,13 @@ function displayWarning(msg) {
 
-  warningTimeout = window.setTimeout(function() {
+  warningTimeout = window.setTimeout(
+    () => {
       warningBox.parentNode.removeChild(warningBox);
       warningTimeout = -1;
-    }, 2000);
+    }, 
+    2000
+  );
```
```js
warningTimeout = setTimeout(() => {
  warningBox.parentNode.removeChild(warningBox);
  warningTimeout = -1;
}, 2000);
```

teoli2003 replied:
Fixed.

Josh-Cena commented on files/en-us/web/api/eventtarget/addeventlistener/index.md:
```diff
@@ -851,7 +851,7 @@ function processEvent(e){
 // Case 3
 for (let i = 0, j = 0 ; i < els.length ; i++){
-  els[j].addEventListener("click", processEvent = function(e){/* do something */}, false);
+  els[j].addEventListener("click", processEvent = (e) => {/* do something */}, false);
```
This looks like a really weird example...

teoli2003 replied:
I'm removing cases 3 and 4, it is very very edge cases. Nobody write code like this (or at least should) and I think this is far, far from any real world usage.

Josh-Cena commented on files/en-us/web/api/file_api/using_files_from_web_applications/index.md:
```diff
@@ -237,7 +237,7 @@ function handleFiles(files) {
 
     const reader = new FileReader();
-    reader.onload = (function(aImg) { return function(e) { aImg.src = e.target.result; }; })(img);
+    reader.onload = ((aImg) => { return (e) => { aImg.src = e.target.result; }; })(img);
```
```suggestion
    reader.onload = ((aImg) => (e) => { aImg.src = e.target.result; })(img);
```

Josh-Cena commented on files/en-us/web/api/history_api/example/index.md:
```diff
@@ -195,7 +195,7 @@ This is an example of an AJAX website composed only of three pages (_first_page.
 "use strict";
 
-const ajaxRequest = new (function () {
+const ajaxRequest = new (() => {
```
This is a constructor. _Maybe_ extract it to a separate declaration?

teoli2003 replied:
The whole example is difficult to understand, not sure I want to go in this rabbit hole now. I'm just reverting.
```suggestion
const ajaxRequest = new (function () {
```

Josh-Cena reviewed (changes requested):
Whew, finally made it through... I have a feeling there are some `this` issues that we didn't catch, because I already spotted one!

teoli2003 commented on files/en-us/web/api/datatransferitem/webkitgetasentry/index.md:
```diff
@@ -114,8 +114,8 @@ function scanFiles(item, container) {
-    directoryReader.readEntries(function(entries) {
-        entries.forEach(function(entry) {
+    directoryReader.readEntries((entries) => {
+        entries.forEach((entry) => {
           scanFiles(entry, directoryContainer);
```
```suggestion
      entries.forEach((entry) => {
        scanFiles(entry, directoryContainer);
```

teoli2003 committed:
Update files/en-us/web/api/datatransferitem/webkitgetasentry/index.md

teoli2003 committed:
Update files/en-us/web/api/datatransferitem/webkitgetasentry/index.md

Co-authored-by: Joshua Chen <sidachen2003@gmail.com>

teoli2003 commented on files/en-us/web/api/domtokenlist/foreach/index.md:
```diff
@@ -62,7 +62,7 @@ const classes = pre.classList;
 classes.forEach(
-  function(value, key, listObj) {
+  (value, key, listObj) => {
     pre.textContent += `(${value} ${key})/${this}\n`;
```
I think both `classes` and `listObj` are the same, so I use the parameter here.
```suggestion
    pre.textContent += `(${value} ${key})/${listObj}\n`;
```

teoli2003 committed:
Update files/en-us/web/api/domtokenlist/foreach/index.md

teoli2003 committed:
Prettify

teoli2003 committed:
Update files/en-us/web/api/file_api/using_files_from_web_applications/index.md

Co-authored-by: Joshua Chen <sidachen2003@gmail.com>

teoli2003 committed:
Revert over-correction

teoli2003 committed:
Remove case 3 and 4

teoli2003 commented:
>  I have a feeling there are some `this` issues that we didn't catch, because I already spotted one!

I'm making an extra pass to double-check this, and request a review when I'm done.

teoli2003 committed:
Remove `this` (was in a polyfill)

teoli2003 commented on files/en-us/web/api/eventtarget/addeventlistener/index.md:
````diff
@@ -606,14 +606,14 @@ a reference to the listener around so you can remove it later.
 
 ```js
-const Something = function(element) {
+const Something = (element) => {
````
This is a constructor, I'm reverting it to:
```suggestion
const Something = function (element) {
```

teoli2003 commented on files/en-us/web/api/eventtarget/addeventlistener/index.md:
````diff
@@ -630,10 +630,10 @@ Another solution is using a special function called `handleEvent()` to catch
 
 ```js
-const Something = function(element) {
+const Something = (element) => {
````
Same here:
```suggestion
const Something = function (element) {
```

teoli2003 commented on files/en-us/web/api/eventtarget/addeventlistener/index.md:
```diff
@@ -710,7 +710,7 @@ As mentioned [above](#specifying_this_using_bind), you can use
 const someString = 'Data';
 
-myButton.addEventListener('click', function () {
+myButton.addEventListener('click', () => {
   console.log(this); // Expected Value: 'Data'
 }.bind(someString));
```
We need to revert here, I think the bind will not work with an arrow function.
```suggestion
myButton.addEventListener('click', function () {
  console.log(this); // Expected Value: 'Data'
}.bind(someString));
```

teoli2003 commented on files/en-us/web/api/eventtarget/addeventlistener/index.md:
```diff
@@ -630,10 +630,10 @@ Another solution is using a special function called `handleEvent()` to catch
   // |this| is a newly created object
   this.name = 'Something Good';
-  this.handleEvent = function(event) {
+  this.handleEvent = (event) => {
```
```suggestion
  this.handleEvent = function (event) {
```

teoli2003 commented on files/en-us/web/api/eventtarget/addeventlistener/index.md:
```diff
@@ -606,14 +606,14 @@ a reference to the listener around so you can remove it later.
   // |this| is a newly created object
   this.name = 'Something Good';
-  this.onclick1 = function(event) {
+  this.onclick1 = (event) => {
```
```suggestion
  this.onclick1 = function (event) {
```

teoli2003 commented on files/en-us/web/api/eventtarget/addeventlistener/index.md:
```diff
@@ -606,14 +606,14 @@ a reference to the listener around so you can remove it later.
   };
 
-  this.onclick2 = function(event) {
+  this.onclick2 = (event) => {
```
```suggestion
  this.onclick2 = function (event) {
```

teoli2003 commented on files/en-us/web/api/eventtarget/addeventlistener/index.md:
````diff
@@ -553,7 +553,7 @@ the element. It will be the same as the value of the `currentTarget` property of
 
 ```js
-my_element.addEventListener('click', function (e) {
+my_element.addEventListener('click', (e) => {
````
```suggestion
my_element.addEventListener('click', function (e) {
```

teoli2003 commented on files/en-us/web/api/eventtarget/eventtarget/index.md:
```diff
@@ -41,7 +41,7 @@ class MyEventTarget extends EventTarget {
 let value = myEventTarget.secret;  // === 5
-myEventTarget.addEventListener("foo", function(e) {
+myEventTarget.addEventListener("foo", (e) => {
   this._secret = e.detail;
```
```suggestion
  myEventTarget._secret = e.detail;
```

teoli2003 commented on files/en-us/web/api/file_api/using_files_from_web_applications/index.md:
```diff
@@ -131,7 +131,7 @@ The code that handles the `click` event can look like this:
 const fileSelect = document.getElementById("fileSelect"),
   fileElem = document.getElementById("fileElem");
```
```suggestion
const fileSelect = document.getElementById("fileSelect");
const fileElem = document.getElementById("fileElem");
```

teoli2003 commented on files/en-us/web/api/htmlinputelement/stepup/index.md:
````diff
@@ -159,7 +159,7 @@ Click the button in this example to increment the {{HTMLElement("input/number",
 ```js
 /* make the button call the function */
 let button = document.getElementById('theButton')
````
```suggestion
const button = document.getElementById('theButton');
```

teoli2003 committed:
const + ;

teoli2003 commented:
I made a pass for `this` and found:

- A case in a polyfill. I deleted the polyfill.
- Several cases in `addEventListener()` that were demonstrating the binding of `this`. I reverted these.
- One actual case of `this` that I fixed.

teoli2003 committed:
const

teoli2003 committed:
Fix this in EventTarget

teoli2003 committed:
Update files/en-us/web/api/eventtarget/addeventlistener/index.md

teoli2003 committed:
Update files/en-us/web/api/eventtarget/addeventlistener/index.md

teoli2003 committed:
Update files/en-us/web/api/eventtarget/addeventlistener/index.md

teoli2003 committed:
Update files/en-us/web/api/eventtarget/addeventlistener/index.md

teoli2003 committed:
Update files/en-us/web/api/eventtarget/addeventlistener/index.md

teoli2003 committed:
Update files/en-us/web/api/eventtarget/addeventlistener/index.md

teoli2003 committed:
Update files/en-us/web/api/eventtarget/addeventlistener/index.md

Josh-Cena commented on files/en-us/web/api/document/getelementsbyclassname/index.md:
````diff
@@ -78,7 +78,7 @@ we'll find all div elements that have a class of 'test':
 ```js
 const testElements = document.getElementsByClassName('test');
-const testDivs = Array.prototype.filter.call(testElements, function(testElement){
+const testDivs = Array.prototype.filter.call(testElements, (testElement) => {
   return testElement.nodeName === 'DIV';
 });
````
```suggestion
const testDivs = Array.prototype.filter.call(
  testElements,
  (testElement) => testElement.nodeName === 'DIV',
);
```

Josh-Cena commented on files/en-us/web/api/event/composedpath/index.md:
````diff
@@ -79,7 +79,7 @@ We then insert one of each element into our page:
 
 ```js
-document.querySelector('html').addEventListener('click',function(e) {
+document.querySelector('html').addEventListener('click',(e) => {
````
```suggestion
document.querySelector('html').addEventListener('click', (e) => {
```

Josh-Cena commented on files/en-us/web/api/eventtarget/addeventlistener/index.md:
```diff
@@ -840,31 +840,6 @@ definition is unchanging, but it is still being repeatedly defined as if new. So
 be multiple identical event listeners, in both cases each iteration will instead
 create a new listener with its own unique reference to the handler function.
 
```
The text above talks about "problem-cases below" but they are non-existent. A rephrase needed.

teoli2003 replied:
Sentence removed

Josh-Cena commented on files/en-us/web/api/file_api/using_files_from_web_applications/index.md:
```diff
@@ -237,7 +237,7 @@ function handleFiles(files) {
 
     const reader = new FileReader();
-    reader.onload = (function(aImg) { return function(e) { aImg.src = e.target.result; }; })(img);
+    reader.onload = ((aImg) => (e) => { aImg.src = e.target.result; })(img);
```
```suggestion
    reader.onload = (e) => { aImg.src = e.target.result; };
```

Closure no longer needed with lexical declarations

teoli2003 committed:
Update files/en-us/web/api/document/getelementsbyclassname/index.md

Co-authored-by: Joshua Chen <sidachen2003@gmail.com>

teoli2003 committed:
Update files/en-us/web/api/event/composedpath/index.md

Co-authored-by: Joshua Chen <sidachen2003@gmail.com>

teoli2003 committed:
Update files/en-us/web/api/file_api/using_files_from_web_applications/index.md

Co-authored-by: Joshua Chen <sidachen2003@gmail.com>

teoli2003 committed:
Remove now useless sentence

Josh-Cena commented on files/en-us/web/api/htmlcanvaselement/mozfetchasstream/index.md:
```diff
@@ -66,8 +66,8 @@ const netutilCallback = function() {
     };
 }
```
For these sorts of examples, what about `netutilCallback = () => (result) => ...`? It's definitely readable to me, but I don't represent general readers...

Josh-Cena replied:
Actually we can talk about that later, when we can enable the `arrow-body-style` rule.

teoli2003 replied:
I've not consistently changed these and accepted both ways. As you are saying, we can decide this at some point and uniformize then.

teoli2003 committed:
Revert to regular function

Josh-Cena commented on files/en-us/web/api/domtokenlist/foreach/index.md:
```diff
@@ -62,8 +62,8 @@ const classes = pre.classList;
-  function(value, key, listObj) {
-    pre.textContent += `(${value} ${key})/${this}\n`;
+  function (value, key, listObj) {
+    pre.textContent += `(${value} ${key})/${listObj}\n`;
```
```suggestion
    pre.textContent += `(${value} ${key})/${this}\n`;
```

Josh-Cena committed:
Update files/en-us/web/api/domtokenlist/foreach/index.md

Josh-Cena reviewed (approved):
I think this is good now!
